docs/guides/testing.mdx

```diff
@@ -38,6 +38,34 @@ The way you mock a dependency depends on your test runner/library.
 
 In [jest](https://jestjs.io/), you can create a `__mocks__/zustand.js` and place the code in that file. If your app is using `zustand/vanilla` instead of `zustand`, then you'll have to place the above code in `__mocks__/zustand/vanilla.js`.
 
+### TypeScript Usage
+
+If you are using zustand according to [TypeScript Guide](./typescript.md), use the following code.
+
+```jsx
+import actualCreate from 'zustand'
+// const actualCreate = jest.requireActual('zustand') // if using jest
+import { act } from 'react-dom/test-utils'
+
+// a variable to hold reset functions for all stores declared in the app
+const storeResetFns = new Set()
+
+// when creating a store, we get its initial state, create a reset function and add it in the set
+const create = () => (createState) => {
+  const store = actualCreate(createState)
+  const initialState = store.getState()
+  storeResetFns.add(() => store.setState(initialState, true))
+  return store
+}
+
+// Reset all stores after each test run
+beforeEach(() => {
+  act(() => storeResetFns.forEach((resetFn) => resetFn()))
+})
+
+export default create
+```
```

Comment:
I need to test this locally, but can I confirm this is essentially identical to the above example, the only change is line 54?